Android/app/src/main/java/app/intra/net/doh/ServerConnectionFactory.java

```diff
@@ -52,13 +57,28 @@ public static String getIpString(Context context, String url) {
     Resources res = context.getResources();
     String[] urls = res.getStringArray(R.array.urls);
     String[] ips = res.getStringArray(R.array.ips);
+    String ret = "";
     for (int i = 0; i < urls.length; ++i) {
       // TODO: Consider relaxing this equality condition to a match on just the domain.
       if (urls[i].equals(url)) {
-        return ips[i];
+        ret = ips[i];
+        break;
+      }
+    }
+
+    try {
+      String domain = new URL(url).getHost();
+      String extraIPs = RemoteConfig.getExtraIPs(domain);
+      if (ret.isEmpty()) {
+        ret = extraIPs;
+      } else if (!extraIPs.isEmpty()) {
+        ret += "," + extraIPs;
```

Comment:
Explain the output format in the method comment.

Reply:
Done